state/state.go

```diff
@@ -295,6 +296,39 @@ func (state *HelmState) Clean() []error {
 	return nil
 }
 
+// FilterReleases allows for the execution of helm commands against a subset of the releases in the helmfile.
+func (state *HelmState) FilterReleases(tags []string) error {
+	var filteredReleases []ReleaseSpec
+	for _, tag := range tags {
+		filter, err := ParseTags(tag)
+		if err != nil {
+			return err
+		}
+		for _, release := range state.Releases {
+			// Allow the name of the release to be matched like a tag
+			if release.Tags == nil {
+				release.Tags = map[string]string{}
+			}
+			release.Tags["name"] = release.Name
+			if filter.Match(release) {
+				// Only add a release once
```

Comment:
The functionality is awesome!
If I could ask more, probably you can change the nesting so that there would be less code, plus and a bit efficiency?

```golang
var filters := []ReleaseFilter{}
for _, tag := range tags {
    f, err := ParseTags(tag)
    // snip - error handling
    filters = append(filters, f)
}
for _, r := range state.Releases {
    for _, f := range filters {
        if f.Match(releases) {
            filteredReleases = append(filteredReleases, release)
            continue
        }
    }
}
```

Reply:
👍